engine/storage/src/org/apache/cloudstack/storage/endpoint/DefaultEndPointSelector.java

```diff
@@ -350,6 +359,24 @@ public EndPoint select(DataObject object, StorageAction action) {
                     }
                 }
             }
+
+            //Handle case where the volume is a volume of an expunging system VM and there are
+            //no other system VMs existing in the zone.
+            if (vm != null) {
+                VirtualMachine.Type type = volume.getAttachedVM().getType();
+                if ((type == VirtualMachine.Type.SecondaryStorageVm || type == VirtualMachine.Type.ConsoleProxy) &&
+                        (vm.getState() == State.Expunging || vm.getState() == State.Destroyed)) {
+
+                    List<SecondaryStorageVmVO> ssvms = ssvmDao.listByZoneId(Role.templateProcessor, volume.getDataCenterId());
+                    if (ssvms == null || ssvms.isEmpty()) {
```

Comment:
@ProjectMoon As a sujestion, this conditional (ssvms == null || ssvms.isEmpty()) can be done using the Apache CollectionUtils.isEmpty method (http://commons.apache.org/proper/commons-collections/javadocs/api-release/org/apache/commons/collections4/CollectionUtils.html#isEmpty%28java.util.Collection%29). It is a null safe verification if the List is empty.